bot/bot.py

```diff
@@ -89,9 +92,32 @@ def clear(self) -> None:
         self._recreate()
         super().clear()
 
+    def remove_extensions(self) -> None:
+        """Remove all extensions and Cog to close bot. Copy from discord.py's own `close` for right closing order."""
+        for extension in tuple(self.extensions):
+            try:
+                self.unload_extension(extension)
+            except Exception:
+                pass
+
+        for cog in tuple(self.cogs):
+            try:
+                self.remove_cog(cog)
+            except Exception:
+                pass
+
     async def close(self) -> None:
         """Close the Discord connection and the aiohttp session, connector, statsd client, and resolver."""
-        await super().close()
+        # Remove extensions and cogs before calling super().close() to allow task finish before HTTP session close
```

Comment:
Let's make this more concise. Bonus: it will fit within 100 characters

```suggestion
        # Done before super().close() to allow tasks finish before the HTTP session closes.
```